src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -645,19 +725,133 @@ void ov::npuw::JustInferRequest::run_subrequest_for_success(std::size_t idx, boo
+        //                              ^tail
+        // The block is always compiled to produce nway. If we need a smaller tensor
+        // on the last iteration, the sub-nway will be copied from the input range to
+        // a temporary tensor, and then the sub-nwway range will be copied from the
+        // request's output range.
+
+        std::size_t offset = 0u;
+        for (std::size_t i = 0u; i < spatial.nway_iters; i++, offset += spatial.nway) {
+            // Collect spatial inputs for this offset
+            for (auto&& param : spatial.params) {
+                const auto& iport = comp_model_desc.compiled_model->inputs()[param.idx];
+                r->set_tensor(
+                    iport,
+                    ov::npuw::util::view(m_spatial_io[real_idx].inputs.at(param.idx), param.dim, offset, spatial.nway));
+            }  // for(params)
+
+            // Now set the spatial outputs
+            for (std::size_t out_idx = 0u; out_idx < num_outputs; out_idx++) {
+                const auto& oport = comp_model_desc.compiled_model->outputs()[out_idx];
+                r->set_tensor(oport,
+                              ov::npuw::util::view(m_spatial_io[real_idx].outputs.at(out_idx),
+                                                   spatial.out_dim,
+                                                   offset,
+                                                   spatial.nway));
+            }  // for(outputs)
+
+            // Now run the part
+            r->infer();
+        }  // for(full_nway_times)
+
+        // Now process the tail, if required
+        if (spatial.tail_size) {
+            // Copy the sub-ranges to spatial inputs
+            // NOTE: tails buffers are read from/written to at 0th offset!
+            for (auto&& param : spatial.params) {
+                auto in_view = ov::npuw::util::view(m_spatial_io[real_idx].inputs.at(param.idx),
+                                                    param.dim,
+                                                    offset,
+                                                    spatial.tail_size);
+
+                const auto& iport = comp_model_desc.compiled_model->inputs()[param.idx];
+                auto out_view = ov::npuw::util::view(m_spatial_io[real_idx].input_tails.at(param.idx),
+                                                     param.dim,
+                                                     0,
+                                                     spatial.tail_size);
+
+                in_view->copy_to(out_view._ptr);
+                r->set_tensor(iport, m_spatial_io[real_idx].input_tails.at(param.idx));
+            }  // for(params)
+
+            // Now set the tail tensors
+            for (std::size_t out_idx = 0u; out_idx < num_outputs; out_idx++) {
+                const auto& oport = comp_model_desc.compiled_model->outputs()[out_idx];
+                r->set_tensor(oport, m_spatial_io[real_idx].output_tails.at(out_idx));
```

Comment:
Why outputs only?

Reply:
I'm not sure if remote tensors support strides (as far as I know L0 tensors don't)